pallets/liquid-staking/src/lib.rs

```diff
@@ -489,50 +485,6 @@ pub mod pallet {
             Ok(().into())
         }
 
-        /// Handle staking settlement at the end of an era
-        /// such as getting reward or been slashed on relaychain.
-        #[pallet::weight(<T as Config>::WeightInfo::record_staking_settlement())]
-        #[transactional]
-        pub fn record_staking_settlement(
-            origin: OriginFor<T>,
-            #[pallet::compact] amount: BalanceOf<T>,
-            kind: StakingSettlementKind,
-        ) -> DispatchResultWithPostInfo {
-            T::RelayOrigin::ensure_origin(origin)?;
-            use StakingSettlementKind::*;
-            match kind {
-                Reward => {
-                    ensure!(
-                        amount <= T::MaxRewardsPerEra::get(),
-                        Error::<T>::ExceededMaxRewardsPerEra
-                    );
-                    StakingPool::<T>::try_mutate(|p| -> DispatchResult {
-                        *p = p.checked_add(amount).ok_or(ArithmeticError::Overflow)?;
-                        Ok(())
-                    })?;
-                    // update exchange rate.
-                    let exchange_rate = Rate::checked_from_rational(
-                        StakingPool::<T>::get(),
-                        T::Assets::total_issuance(Self::liquid_currency()?),
-                    )
-                    .ok_or(Error::<T>::InvalidExchangeRate)?;
-                    ExchangeRate::<T>::put(exchange_rate);
```

Comment:
3/3 
while in former logic, we use `StakingPool`, when user unstake, `StakingPool` will decrease too.